### PR TITLE
Update dependencies + add test case for Cache using ETag - see #5

### PR DESCRIPTION
**Pull request by mikespub**
Keep dependencies up to date + show use of ETag in test cases

### CODE REVIEW

**Comment by mikespub** [comment]
Updated composer.json in case you decide to do a major version bump for middleware/utils https://github.com/middlewares/utils/pull/27#discussion_r1428965229

**Comment by mikespub** [comment]
hi @oscarotero I updated the workflow here as well now - see middlewares/utils#27

File: composer.json [hunk 1]
```diff
@@ -14,20 +14,21 @@
         "issues": "https://github.com/middlewares/cache/issues"
     },
     "require": {
-        "php": "^7.2 || ^8.0",
-        "middlewares/utils": "^3.0",
-        "micheh/psr7-cache": "^0.5.0",
-        "psr/cache": "^1.0",
+        "php": ">=8.1",
+        "middlewares/utils": "^3.3 || ^4.0",
```
Comment (oscarotero) [inline composer.json]: This must be a major version, so I think it could require only `^4.0` version.

**Review by oscarotero — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (mikespub) [reply composer.json]: No problem - done.

BTW, there's a typo in https://github.com/middlewares/utils/blob/c789be3ce5465928cedc8d1492172ac36918ede4/CHANGELOG.md?plain=1#L15 - it should read "PHP prior to 8.1" not 8.2 :-)

Comment (oscarotero) [reply composer.json]: hmm, I thought "Prior" means "before", so it removes support for versions before (or prior to) 8.2 (but 8.2 is still supported). Does it make sense? (I'm not english native, so I'm not sure)

Comment (mikespub) [reply composer.json]: Indeed prior means before - but we support 8.1+ here, not 8.2+

Comment (oscarotero) [reply composer.json]: Oh, you're right. I won't release a new version just for this but I've fixed the changelog.md and edited the release description of GitHub to reflect this change. Thanks!

**Pull request merged**